fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt

```diff
@@ -0,0 +1,235 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.DEFAULT_MAX_LINES
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private const val ANIMATION_DURATION_MILLIES = 2000
+
+/**
+ * Download indicator.
+ *
+ * @param text The download indicator text to be displayed.
+ * @param modifier [Modifier] to be applied to the layout.
+ * @param contentDescription to be applied to the layout.
+ * @param backgroundColor The background [Color] of the download indicator.
+ * @param textColor [Color] to apply to the download indicator text.
+ * @param icon Optional [Painter] used to display an [Icon] before the download indicator text.
+ * @param iconContentDescription to be applied to the icon.
+ * @param iconModifier to be applied to the icon.
+ * @param iconTint to be applied to the icon.
+ */
+@Composable
+fun DownloadIndicator(
+    text: String,
+    modifier: Modifier = Modifier,
+    contentDescription: String? = null,
+    backgroundColor: Color = FirefoxTheme.colors.actionPrimary,
+    textColor: Color = FirefoxTheme.colors.textActionPrimary,
+    icon: Painter? = null,
+    iconContentDescription: String? = null,
+    iconModifier: Modifier = Modifier,
+    iconTint: Color = FirefoxTheme.colors.iconActionPrimary,
+) {
+    BaseDownloadIndicator(
+        text = text,
+        textColor = textColor,
+        modifier = modifier,
+        buttonContentDescription = contentDescription,
+        icon = icon,
+        backgroundColor = backgroundColor,
+        iconModifier = iconModifier.then(
+            Modifier.wrapContentSize(),
+        ),
+        iconContentDescription = iconContentDescription,
+        iconTint = iconTint,
+    )
+}
+
+/**
+ * Icon for Download indicator.
+ *
+ * @param modifier [Modifier] to be applied to the icon layout.
+ * @param icon [Painter] used to be displayed.
+ * @param tint to be applied to the icon.
+ * @param iconContentDescription Optional contentDescription for the icon.
+ * */
+@Composable
+fun DownloadIconIndicator(
```

Comment:
I tested this component in TalkBack and I think it may need some accessibility refinement. Currently, TalkBack only says "Translating" on the moving spinner icon.

I looked around in our code base some more and an indeterminate progress bar has recently been added via [shopping](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProgressIndicator.kt#31,41,51).

To test the shopping loading icon feature:
1. Turn on TalkBack
2. Navigate to Walmart.com and select a more uncommon item (I picked things like grocery items)
3. Go to "Open review checker" in toolbar
4. "Check review quality" or something similar for an already processed item
5. Spinner will show up
6. Navigate to spinner
7. Talkback announces:  "In Progress", "Progress Bar". I think "In Progress" is the `<value>` and "Progress Bar" is the `<role>`. I looked in the [layout inspector semantics](https://developer.android.com/jetpack/compose/semantics) and it looks like it inherited from "ProgressBarRangeInfo". (Aside, I think it probably should have became a live region to hear that announcement without needing to manually navigate to it, but it could be my TalkBack settings?)

We used an [indeterminate progress circle indicator](https://developer.android.com/jetpack/compose/components/progress) here to make a [progress indicator](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProgressIndicator.kt#31,41,51).

A quick look at [CircularProgressIndicator](https://developer.android.com/reference/com/google/android/material/progressindicator/CircularProgressIndicator) doesn't seem to have a way to override the icon/animation (could be missing something here, please double check).

I'm thinking we will need to update this component to use [custom semantics](https://developer.android.com/jetpack/compose/accessibility) to make this have the `ProgressBarRangeInfo` [role](https://developer.android.com/reference/kotlin/androidx/compose/ui/semantics/SemanticsProperties#ProgressBarRangeInfo()).

Reply:
@anna-yeddi, does this seem to be the right direction?

Reply:
Done

Reply:
Thank you for all the work on this. I think this solution will meet accessibility criteria with just a few more adjustments. 

So, this patch as it stands is manually overriding everything to provide an accurate screen reader description. However, we are losing the semantic value of the components. I think this might be okay, but needs more discussion.

So originally, I hoped we could do this:
```
        modifier = Modifier.semantics {
            role = Role.ProgressBarRangeInfo
        }
```

([Example](https://bryanherbst.com/2021/02/19/compose-role-semantics/). However, it seems compose does not support this (wasn't in the list of IDE choices when I imported role.)). If this was possible, then we would have gotten a nice inherited `role` and `value` from the Progress Bar.

I also didn't see a way to customize a [`CircularProgressIndicator()`](https://developer.android.com/reference/com/google/android/material/progressindicator/CircularProgressIndicator) for our needs e.g, changing the drawable to our sync icon. 


**I have a question:**
1. I think we should remove `clearAndSetSemantics` or could you provide further justification? That is overriding the semantic meaning of the elements and generally we need special reasoning to do so. I also do not see its use in other places in our code base. 

I'm bringing it up because setting the contentDescription this way via `clearAndSetSemantics` to "Translating in Progress" is losing the `role` of button and that it is not-enabled.

**My recommendation:**
* Please remove `clearAndSetSemantics` - it is overriding too much semantic content, unless there is further justification
* Add a parameter for us to set the `contentDescription` on the elements as a freestanding option 
* Set the default `contentDescription` of the icon to a string of `In progress`.
* Keep the default `contentDescription` of the translating button to a string of `Translating in progress`
*  (We will also need a `Downloading in progress` string in the future for use on the download screen)
* I'll add a note on the Figma about this to see if we get further recommendations and notify Emily of the string need here. (We can return and refine this later as well. I think these recommendations will push it fairly close to where it needs to be.)

Reply:
(Aside note, I think when we add this to the sheet proper, we will want the containing area to be a polite live-region to notify the user of the button change.)

Reply:
Thanks for discussing this issue in Slack with me! 

@iorgamgabriel found a great solution. We are just waiting to get a little more information on the preferred labeling to know which route to take and this patch will be good to go with a quick update.  🎉

Reply:
Have some more info on the labling:

TalkBack should report: "Translating in progress, button, disabled"

via what you shared earlier: 

```
    modifier = modifier.then(
        Modifier
            .clearAndSetSemantics {
                disabled()
                role = Role.Button
                contentDescription?.let { this.contentDescription = contentDescription }
            },
    ),
```

And for the indicator, "Download in progress" as the content description. (More context in the Figma, if needed too.)

With that change and what @gabrielluong mentioned, and we should be good to go.